tests/ops/functions/conftest.py

```diff
@@ -14,74 +14,125 @@
+from pennylane.operation import DiagGatesUndefinedError, MatrixUndefinedError
 from pennylane.ops.op_math.adjoint import Adjoint, AdjointObs, AdjointOperation, AdjointOpObs
 
-# if you would like to validate one of these operators, add an instance to the parametrization
-# of `test_explicit_list_of_ops` in `test_assert_valid.py`, and kindly move it below the
-# "manually validated ops" comment in this dict for easy inspection in the future.
-_SKIP_OP_TYPES = {
-    # ops composed of more than one thing
-    Adjoint,
-    Tensor,
-    qml.Hamiltonian,
-    qml.ops.Pow,
-    qml.ops.SProd,
-    qml.ops.Prod,
-    qml.ops.Controlled,
-    qml.ops.Exp,
-    qml.ops.Evolution,
-    qml.ops.Conditional,
-    # fails for unknown reason - should be registered in the test parametrization below
-    qml.SparseHamiltonian,  # fails because data is not as expected
-    qml.pulse.ParametrizedEvolution,
-    qml.THermitian,
-    qml.resource.FirstQuantization,
-    qml.SpecialUnitary,
-    qml.IntegerComparator,
-    qml.PauliRot,
-    qml.PauliError,
-    qml.ops.qubit.special_unitary.TmpPauliRot,  # private object
-    qml.StatePrep,
-    qml.PCPhase,
-    qml.BlockEncode,
-    qml.resource.DoubleFactorization,
-    qml.resource.ResourcesOperation,
-    # templates
-    *[i[1] for i in getmembers(qml.templates) if isclass(i[1]) and issubclass(i[1], Operator)],
-    # manually validated ops
-    qml.ops.Sum,
-    qml.BasisState,
-    qml.ControlledQubitUnitary,
-    qml.QubitStateVector,
-    qml.QubitChannel,
-    qml.MultiControlledX,
-}
+_INSTANCES_TO_TEST = [
+    qml.sum(qml.PauliX(0), qml.PauliZ(0)),
+    qml.BasisState([1], wires=[0]),
+    qml.ControlledQubitUnitary(np.eye(2), control_wires=1, wires=0),
+    qml.QubitStateVector([0, 1], wires=0),
+    qml.QubitChannel([np.array([[1, 0], [0, 0.8]]), np.array([[0, 0.6], [0, 0]])], wires=0),
+    qml.MultiControlledX(wires=[0, 1]),
+    qml.Projector([1], 0),  # the state-vector version is already tested
+    qml.SpecialUnitary([1, 1, 1], 0),
+    qml.IntegerComparator(1, wires=[0, 1]),
+    qml.PauliRot(1.1, "X", wires=[0]),
+    qml.StatePrep([0, 1], 0),
+    qml.PCPhase(0.27, dim=2, wires=[0, 1]),
+    qml.BlockEncode([[0.1, 0.2], [0.3, 0.4]], wires=[0, 1]),
+    qml.adjoint(qml.PauliX(0)),
+    qml.adjoint(qml.RX(1.1, 0)),
+    Tensor(qml.PauliX(0), qml.PauliX(1)),
+    qml.Hamiltonian([1.1, 2.2], [qml.PauliX(0), qml.PauliZ(0)]),
+    qml.s_prod(1.1, qml.RX(1.1, 0)),
+    qml.prod(qml.PauliX(0), qml.PauliY(1), qml.PauliZ(0)),
+    qml.ctrl(qml.RX(1.1, 0), 1),
+    qml.exp(qml.PauliX(0), 1.1),
+    qml.ops.Evolution(qml.PauliX(0), 5.2),
+    qml.QutritBasisState([1, 2, 0], wires=[0, 1, 2]),
+]
+"""Valid operator instances that could not be auto-generated."""
+
+
+_INSTANCES_TO_FAIL = [
+    (
+        qml.SparseHamiltonian(qml.Hamiltonian([1.1], [qml.PauliX(0)]).sparse_matrix(), [0]),
+        AssertionError,  # each data element must be tensorlike
+    ),
```

Comment:
What would you think of expanding the definition of `TensorLike` to include `scipy.sparse.sparray` and `scipy.sparse.spmatrix`? Seems to work out for me locally. 

We then run into the issue that `allclose` doesn't work for sparse mats, but we can probably fix that too.

Reply:
I'm ok with it, but it will probably take more work as you're suggesting with the benefit pretty much just being us validating `SparseHamiltonian`